front/src/modules/pathfinding/hook/usePathfinding.ts

```diff
@@ -0,0 +1,332 @@
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+import useInfra from './useInfra';
+
+export function reducer(state: PathfindingState, action: PathfindingActionV2): PathfindingState {
+  switch (action.type) {
+    case 'PATHFINDING_STARTED': {
+      return {
+        ...state,
+        running: true,
+        done: false,
+        error: '',
+        mustBeLaunched: false,
+        cancelled: false,
+      };
+    }
+    case 'PATHFINDING_CANCELLED': {
+      return {
+        ...state,
+        running: false,
+        done: false,
+        error: '',
+        mustBeLaunched: false,
+        cancelled: true,
+      };
+    }
+    case 'PATHFINDING_FINISHED': {
+      if (state.cancelled) {
+        return {
+          ...state,
+          running: false,
+          done: false,
+          error: '',
+          mustBeLaunched: false,
+          mustBeLaunchedManually: false,
+        };
+      }
+      return {
+        ...state,
+        running: false,
+        done: true,
+        error: '',
+        mustBeLaunched: false,
+        mustBeLaunchedManually: false,
+      };
+    }
+    case 'PATHFINDING_ERROR': {
+      return {
+        ...state,
+        running: false,
+        done: false,
+        error: action.message || '',
+        mustBeLaunched: false,
+      };
+    }
+    case 'PATHFINDING_PARAM_CHANGED':
+    case 'VIAS_CHANGED':
+    case 'INFRA_CHANGED': {
+      if (
+        !action.params ||
+        state.running ||
+        (action.type === 'INFRA_CHANGED' && !action.params.origin && !action.params.destination)
+      ) {
+        return { ...state };
+      }
+      const { origin, destination, rollingStock } = action.params;
+      if (!origin || !destination || !rollingStock) {
+        return {
+          ...state,
+          running: false,
+          error: '',
+          done: false,
+          missingParam: true,
+        };
+      }
+      return {
+        ...state,
+        error: '',
+        mustBeLaunched: true,
+        missingParam: false,
+      };
+    }
+    default:
+      throw new Error('Pathfinding action doesn’t exist');
+  }
+}
+
+function init({
+  origin,
+  destination,
+  rollingStock,
+  pathSteps,
+  pathProperties,
+}: {
+  origin: PathStep | null;
+  destination: PathStep | null;
+  rollingStock?: RollingStockWithLiveries;
+  pathSteps: (PathStep | null)[];
+  pathProperties?: ManageTrainSchedulePathProperties;
+}): PathfindingState {
+  if (compact(pathSteps).length === 0 || !pathProperties?.geometry) {
+    return {
+      ...initialState,
+      mustBeLaunched: Boolean(origin) && Boolean(destination) && Boolean(rollingStock),
+    };
+  }
+  return initialState;
+}
+
+type PathfindingProps = {
+  pathProperties?: ManageTrainSchedulePathProperties;
+  setPathProperties: (pathProperties?: ManageTrainSchedulePathProperties) => void;
+};
+
+export default function usePathfindingV2({ pathProperties, setPathProperties }: PathfindingProps) {
```

Comment:
To avoid any confusion, I think we should keep the "props" term only for react components. A custom hook is just a function where we can use hooks so parameters should be passed like a regular function.

```suggestion
const usePathfindingV2 = (
  setPathProperties: (pathProperties?: ManageTrainSchedulePathProperties) => void,
  pathProperties?: ManageTrainSchedulePathProperties
) => {
```